### PR TITLE
fix: alloy-rs/core update

### PR DESCRIPTION
**Pull request by refcell**
### Description

- Fixes `op-alloy` for alloy-rs/core `no_std` update.
- Adds `op-alloy-protocol` to the `no_std` ci job matrix.

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (emhane) [inline .github/workflows/ci.yml:1]: these changes are unrelated, but necessary, pls open sep pr

**Review by emhane — CHANGES_REQUESTED** [review]
all these types are redundant imo, they already exists in `alloy` or `op-alloy` or variations thereof

File: .github/workflows/ci.yml
Comment (refcell) [reply .github/workflows/ci.yml:1]: For sure, will actually just remove the redundant types for this pr and isolate the changes :)

Comment (refcell) [reply .github/workflows/ci.yml:1]: Thanks for the review - I had a hunch these types were redundant, just didn't know where they lived.

File: crates/protocol/src/block.rs [hunk 1]
```diff
@@ -1,13 +1,71 @@
...
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub enum BlockKind {
+    /// The earliest known block.
+    Earliest,
+    /// The latest pending block.
+    Latest,
+    /// The latest finalized block.
+    Finalized,
+}
```
[the remark this answers is not in the document]
Comment (refcell) [reply crates/protocol/src/block.rs]: We actually don't need this - was legacy code, removed.

File: crates/protocol/src/block.rs [hunk 2]
```diff
@@ -1,13 +1,71 @@
...
+pub struct Block {
+    /// Block header.
+    pub header: Header,
+    /// Transactions in this block.
+    pub body: Vec<TxEnvelope>,
+    /// Ommers/uncles header.
+    pub ommers: Vec<Header>,
+    /// Block withdrawals.
+    pub withdrawals: Option<Vec<Withdrawal>>,
+}
```
[the remark this answers is not in the document]
Comment (refcell) [reply crates/protocol/src/block.rs]: Will work on making this compatible.

File: crates/protocol/src/block.rs [hunk 3]
```diff
@@ -1,13 +1,71 @@
...
+pub struct OpBlock {
+    /// Block header.
+    pub header: Header,
+    /// Transactions in this block.
+    pub body: Vec<OpTxEnvelope>,
+    /// Ommers/uncles header.
+    pub ommers: Vec<Header>,
+    /// Block withdrawals.
+    pub withdrawals: Option<Vec<Withdrawal>>,
+}
```
[the remark this answers is not in the document]
Comment (refcell) [reply crates/protocol/src/block.rs]: Will work on making this compatible.

**Review by emhane — APPROVED** [review]

**Pull request merged**